vertx-core/src/main/java/io/vertx/core/http/HttpClientOptions.java

```diff
@@ -530,6 +538,24 @@ public HttpClientOptions setHttp2KeepAliveTimeout(int keepAliveTimeout) {
     return this;
   }
 
+  /**
+   * @return the HTTP/2 upgrade maximum length of the aggregated content in bytes
+   */
+  public int getHttp2UpgradeMaxContentLength() {
+    return http2UpgradeMaxContentLength;
+  }
+
+  /**
+   * Set the HTTP/2 upgrade maximum length of the aggregated content in bytes.
```

Comment:
I think we need to add that this is only effective when `http2ClearTextUpgradeWithPreflightRequest` is set to false (which is the default).

When http2ClearTextUpgradeWithPreflightRequest is true, then the client makes a preflight OPTIONS request and the upgrade will not send a body, voiding the requirements.

Reply:
Good point!